src/parseFromAnchor.js

```diff
@@ -12,7 +26,7 @@ function parseFromAnchor(opt_uri) {
 		hostname: link.hostname,
 		password: link.password,
 		pathname: link.pathname[0] === '/' ? link.pathname : '/' + link.pathname,
-		port: link.port,
+		port: extractPortFromURI(opt_uri),
```

Comment:
I like it. I just want to clarify the use of the `http://localhost`. We're going to use it as a template only for testing if the browser sets a misleading port, and then, we're going to replace the `link.href` with the given `opt_uri`. Right? Because for some reason the developer could choose to set a `0` port intentionally.